src/libs/NumberUtils.js

```diff
@@ -1,30 +1,39 @@
 import CONST from '../CONST';
 
 /**
- * Generates a random positive 64 bit numeric string by randomly generating the left half and right half and concatenating them. Used to generate client-side ids.
+ * Generates a random positive 64 bit numeric string by randomly generating the left, middle, and right parts and concatenating them. Used to generate client-side ids.
  * @returns {String} string representation of a randomly generated 64 bit signed integer
  */
 /* eslint-disable no-unused-vars */
 function rand64() {
     // Max 64-bit signed:
     // 9,223,372,036,854,775,807
-    // The left part of the max 64-bit number *+1* because we're flooring it
-    const left = Math.floor(Math.random() * (CONST.MAX_64BIT_LEFT_HALF + 1));
+    // The left part of the max 64-bit number *+1* because we're flooring it.
+    const left = Math.floor(Math.random() * (CONST.MAX_64BIT_LEFT_PART + 1));
 
-    // The right part of the max 64-bit number *+1* for the same reason
+    let middle;
     let right;
 
-    // If the top is any number but the highest one, we can actually have any value for the rest
-    if (left !== CONST.MAX_64BIT_LEFT_HALF) {
-        right = Math.floor(Math.random() * 1000000000);
+    // If the left is any number but the highest possible, we can actually have any value for the middle part, because even if it's all `9`s, the final value will not overflow the maximum
+    // 64-bit number.
+    if (left !== CONST.MAX_64BIT_LEFT_PART) {
+        middle = Math.floor(Math.random() * 10000000);
```

Comment:
Could you make `10000000` a constant and set it to the calculation that you used to get this number. Also please explain with a brief comment.

Reply:
Agree, the comment above already explains that it can set the `middle` variable all the way up to `9999999` (any value up to 7 digits) but the `10000000` is a little confusing at first!